dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/FileWriter.kt

```diff
@@ -54,25 +50,23 @@ internal class FileWriter<T : Any>(
             if (data.length >= MAX_LOG_SIZE) {
                 // TODO RUMM-49 warn user that the log is too big !
             } else {
-                obfuscateAndWriteData(data)
+                synchronized(this) {
+                    writeData(data)
+                }
             }
         })
     }
 
-    private fun obfuscateAndWriteData(data: String) {
-        val obfData = obfuscate(data)
-
-        synchronized(this) {
-            writeLogSafely(obfData)
-        }
-    }
-
-    private fun writeLogSafely(obfData: ByteArray) {
+    private fun writeData(data: String) {
         var file: File? = null
         try {
-            file = fileOrchestrator.getWritableFile(obfData.size)
-            file.appendBytes(obfData)
-            file.appendBytes(separator)
+            val dataAsByteArray = data.toByteArray(Charsets.UTF_8)
+            file = fileOrchestrator.getWritableFile(dataAsByteArray.size)
+            val fileSize = file.length()
+            if (fileSize > 0) {
+                file.appendBytes(separator)
```

Comment:
appendByte will create a new FileOutputstream twice in most cases. You can actually do : 
```kotlin
FileOutputStream(this, true).use { 
    if (fileSize > 0) {
        it.write(array) 
    }
    it.write(dataAsByteArray) 
}
```